src/opencensus.erl

```diff
@@ -0,0 +1,118 @@
+%%%-------------------------------------------------------------------
+%% @doc opencensus
+%% @end
+%%%-------------------------------------------------------------------
+-module(opencensus).
+
+-export([start_span/3,
+         finish_span/1,
+         child_span/2,
+
+         timestamp/0,
+         generate_trace_id/0,
+         generate_span_id/0]).
+
+-include("opencensus.hrl").
+
+-export_type([trace_id/0,
+              span_id/0,
+              trace_context/0,
+              span/0]).
+
+-type trace_id()      :: non_neg_integer().
+-type span_id()       :: non_neg_integer().
+-type trace_context() :: #trace_context{}.
+-type span()          :: #span{}.
+
+-type annotations()   :: maps:map(unicode:chardata(), attribute_value()).
```

Comment:
Using maps will make this not usable in older Erlang versions, not a big issue to me but it might be worth considering and making a conscious decision to not support older Erlang.

Reply:
Yea, I think supporting 4 major versions is plenty :)